server/backend/database/database.go

```diff
@@ -163,6 +164,20 @@ type Database interface {
 		candidatesLimit int,
 	) ([]*ClientInfo, error)
 
+	// FindDocumentHardDeletionCandidatesPerProject finds the documents that need to be deleted housekeeping per project.
+	FindDocumentHardDeletionCandidatesPerProject(
+		ctx context.Context,
+		project *ProjectInfo,
+		candidatesLimit int,
+		documentHardDeletionGracefulPeriod gotime.Duration,
+	) ([]*DocInfo, error)
+
+	// DeleteDocument Document complete deletion in progress
+	DeleteDocument(
```

Comment:
How about changing this function name to `DeleteDocuments()`, since we are receiving multiple documents?